src/components/Game.tsx

```diff
@@ -114,10 +119,24 @@ const Game = () => {
   const collectedItems = useGameStore((state) => state.collectedItems);
   const bonkBackTo = useGameStore((state) => state.bonkBackTo);
 
+  // const { getCurrentViewport } = useThree((state) => state.viewport);
+
   const marbleRef = useRef<Mesh>(null);
   const orbit = useRef<OrbitControlsImpl>(null);
   const { camera, viewport } = useThree();
 
+  const level = useMemo(() => {
+    if (currentLevelId) {
+      return levels.find((l) => l.id === currentLevelId);
+    }
+  }, [levels, currentLevelId]);
+
+  const currentTile = useMemo(() => {
+    if (currentTileId && level) {
+      return level.tiles.find((t): t is TrackTile => t.id === currentTileId);
+    }
+  }, [currentTileId, level]);
```

Comment:
this bug sucked: current tile stored in memory was not updated when its underlying tile was.